build-system/server/app.js

```diff
@@ -56,8 +56,9 @@ app.use(bodyParser.text());
 
 // Middleware is executed in order, so this must be at the top.
 // TODO(#24333): Migrate all server URL handlers to new-server/router and
-// deprecate this file.
-if (argv.new_server) {
+// deprecate app.js.
+// TODO(erwinmombay, #32865): Make visual diff tests use the new server
+if (!argv._.includes('visual-diff')) {
```

Comment:
how do we want to handle this in the `gulp serve` case?

Reply:
With this change, `gulp serve` will always build the new transforms, and developers may use them if they wish by passing in `--esm`. For the visual diff tests, I think you'll need to change how the visual test pages consume URLs, at which point we may be able to remove this `if` guard and always use `new-server/router`.